.github/workflows/emily-cron-ecr-build.yaml

```diff
@@ -0,0 +1,150 @@
+name: Build Docker Image for Emily Cron
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment name"
+        required: true
+        default: environment/sbtc-devnet
+        type: choice
+        options:
+          - environment/sbtc-devnet
+          - environment/sbtc-private-mainnet
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: docker-build-emily-cron-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+  attestations: write
+  packages: write
+  issues: read
+  pull-requests: read
+
+env:
+  docker_platforms: "linux/amd64"
+
+jobs:
+  resolve-env:
+    name: Resolve Environment
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.set_env.outputs.environment }}
+      should_run: ${{ steps.set_env.outputs.should_run }}
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Resolve Environment
```

Comment:
sanity check: if we use the labels to trigger a workflow, using a single tag it will run both the build images and the Emily deployment. It's true that we could then just reject the workflow run when it ask for an approval, but I wanted to double check that this _run everything related when we specify (eg) `sbtc-devnet`_ is what we want